daemon/mgr/container_envs.go

```diff
@@ -0,0 +1,181 @@
+package mgr
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/alibaba/pouch/pkg/utils"
+
+	"github.com/magiconair/properties"
+	"github.com/sirupsen/logrus"
+)
+
+// container env is used to manage environment variables in container.
+// In original ways, processes in container can only accepted env when startup.
+// And container engine would store these data in container.json which is persistent on local disk.
+// There is no env ralated data in container or image.
+//
+// While for rich container mode, env management in container is a little bit complicated.
+// First, introduce some scenarios in practice:
+// * init process in rich container will not pass envs to user-specific CMD process if user switched from root;
+// * newly created shell command should herit all envs of the container, so all envs should be stored in dir
+// /etc/profild.d/*.sh (here is pouchenv.sh). After that a process execed from outside would take env of init process,
+// then such action like reload of detailed user application could share the env.
+const (
+	// containerInstanceInfo is file in the container image.
+	// row in this file has content like `env_appDeployType = taoJavaWeb`
+	containerInstanceInfo = "/etc/instanceInfo"
+	// pouchEnvDir =
+	pouchEnvDir = "/etc/profile.d"
+	// this filepath is used in PouchContainer to store user input env via persistent file
+	pouchEnvFile = "/etc/profile.d/pouchenv.sh"
+)
+
+// updateContainerEnv update the container's envs in
+// /etc/instanceInfo and /etc/profile.d/pouchenv.sh
+// Env used by rich container.
+func updateContainerEnv(inputRawEnv []string, baseFs string) error {
+	var (
+		envPropertiesPath = path.Join(baseFs, containerInstanceInfo)
+		envShPath         = path.Join(baseFs, pouchEnvFile)
+	)
+
+	// check the existence of related files.
+	// if dir of pouch.sh is not exist, it's unnecessary to update that files.
+	if _, err := os.Stat(path.Join(baseFs, pouchEnvDir)); err != nil {
```

Comment:
i think we can delete this part code. WDYT?

Reply:
I do not think so. Since check dir, return nil if err; check file, return err if err.